src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -455,9 +455,16 @@ class DistributedClosestPointImpl
     axom::Array<axom::IndexType> domIds(ptCount, ptCount);
     std::size_t copiedCount = 0;
     conduit::Node tmpValues;
-    for(conduit::index_t d = 0; d < mdMeshNode.number_of_children(); ++d)
+    for(axom::IndexType d = 0; d < mdMeshNode.number_of_children(); ++d)
```

Comment:
I looked into this recently and `number_of_children()` returns a `conduit::index_t`, which is a signed int (either 32-bits or 64-bits based on how conduit is configured)

`axom::IndexType` is also a signed int32 or int64, based on how axom is configured.
However, there's no guarantee that `conduit::index_t` has the same bitwidth as `axom::IndexType`.

In practice, we're unlikely to run problems with billions of ranks, but if we're explicitly casting to a 32-bit int a few lines down, we should probably be consistent throughout this loop.

See: 
* https://github.com/LLNL/conduit/blob/f570d0dc4a77861ea129b2f5965768f77ce68030/src/libs/conduit/conduit_bitwidth_style_types.h.in#L330-L336
* https://github.com/LLNL/conduit/blob/f570d0dc4a77861ea129b2f5965768f77ce68030/src/libs/conduit/conduit_core.hpp#L63-L68
* https://github.com/LLNL/axom/blob/f511a2593369043c2571f0646fb0ce421acbc572/src/axom/core/Types.hpp#L63-L67